### PR TITLE
Add HazardousMaterials to Measure

### PR DESCRIPTION
**Pull request by macintoshpie**

### CODE REVIEW

**Review by markborkum — CHANGES_REQUESTED** [review]
Asserting the existence of given a hazard does not identify said hazard, which the UN recommends.

Consider replacing the 0..1 occurrences, boolean-valued element with a 0..N occurrences, enumeration-valued element, where the enumerations are the hazard categories from the Globally Harmonized System of Classification and Labelling of Chemicals (GHS): "GHS01: Explosive", "GHS02: Flammable", etc. [1]

[1] https://en.wikipedia.org/wiki/GHS_hazard_pictograms

**Comment by corymosiman12** [comment]
I would suggest keeping the boolean as a flag, then have the enumerated aspect as well.

**Review by corymosiman12 — CHANGES_REQUESTED** [review]
Same as Mark

**Comment by macintoshpie** [comment]
I decided not to keep the boolean `HazardousMaterialsPresent` because I think it wouldn't be adding anything useful

**Review by markborkum — CHANGES_REQUESTED** [review]
Please rename the new element to `auc:HazardPictogram` to match the naming convention in the GHS documentation.

Also, if this *is* a GHS hazard pictogram, then there must not be an "Other" enumeration value. The complete list of GHS hazard pictograms is available at https://en.wikipedia.org/wiki/GHS_hazard_pictograms (including transport pictograms, etc.).

File: BuildingSync.xsd [hunk 1]
```diff
@@ -5758,6 +5758,34 @@
...
       </xs:element>
+      <xs:element name="HazardousMaterials" minOccurs="0">
+        <xs:annotation>
+          <xs:documentation>List of the hazardous materials determined to be present, either through observation or as reported by others, which could potentially have a significant impact on the measure cost estimates.</xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="HazardousMaterial" minOccurs="0" maxOccurs="unbounded">
+              <xs:annotation>
+                <xs:documentation>Hazard types, defined by the Globally Harmonized System of Classification and Labelling of Chemicals (GHS).</xs:documentation>
```
Comment (nllong) [inline BuildingSync.xsd:5769]: @macintoshpie -- also, use labeling not labelling (former is US spelling, latter is UK).

Comment (nllong) [reply BuildingSync.xsd:5769]: https://www.osha.gov/dsg/hazcom/ghsguideoct05.pdf

**Review by markborkum — CHANGES_REQUESTED** [review]
To comply with the standard and to interoperate with preexisting chemical health and safety software, the set of enumeration values must exactly match the GHS documentation, i.e., "GHS01: Explosive", ..., "GHS09: Environmental hazard", "Divisions 1.1–1.3", etc.

**Comment by macintoshpie** [comment]
@markborkum I'm not seeing those prefixes in the document Nick shared, are you looking at another document by GHS?

**Comment by markborkum** [comment]
@macintoshpie In chemical information management systems and laboratory information management systems, it is best practice to include the "label element", "symbol" and "signal word" in enumeration values for GHS hazard pictograms (for example, look at the alt-text for the GHS hazard pictogram in the infobox on this Wikipedia page: https://en.wikipedia.org/wiki/Methane). Older CIMS and LIMS may reference GHS hazard pictograms by only the label element or symbol, so including the additional data facilitates data integration.

The canonical reference for the GHS is the Purple Book: https://www.unece.org/trans/danger/publi/ghs/ghs_rev07/07files_e.html

The most recent revision is available at: https://www.unece.org/fileadmin/DAM/trans/danger/publi/ghs/ghs_rev07/English/ST_SG_AC10_30_Rev7e.pdf

**Comment by macintoshpie** [comment]
@markborkum I still can't find those prefixes/enumerations in the document you shared. Also, do we necessarily want to include pictogram information if we're describing hazard categories defined by GHS? It seems that the pictograms are image specific, and don't cover all of the cases mentioned in the documents

Could you take a stab at adding these IDs since I'm not sure where to source them?

**Comment by markborkum** [comment]
@macintoshpie 

```xml
<xs:element name="HazardPictogram" minOccurs="0" maxOccurs="unbounded">
  <xs:annotation>
    <xs:documentation>Hazard pictograms, defined by the Globally Harmonized System of Classification and Labelling of Chemicals (GHS).</xs:documentation>
  </xs:annotation>
  <xs:simpleType>
    <xs:restriction base="xs:string">
      <xs:enumeration value="GHS01: Explosive"/>
      <xs:enumeration value="GHS02: Flammable"/>
      <xs:enumeration value="GHS03: Oxidizing"/>
      <xs:enumeration value="GHS04: Compressed Gas"/>
      <xs:enumeration value="GHS05: Corrosive"/>
      <xs:enumeration value="GHS06: Toxic"/>
      <xs:enumeration value="GHS07: Harmful"/>
      <xs:enumeration value="GHS08: Health hazard"/>
      <xs:enumeration value="GHS09: Environmental hazard"/>
      <xs:enumeration value="Division 1.4"/>
      <xs:enumeration value="Division 1.5"/>
      <xs:enumeration value="Division 1.6"/>
      <xs:enumeration value="Division 2.1"/>
      <xs:enumeration value="Division 2.2"/>
      <xs:enumeration value="Division 2.3"/>
      <xs:enumeration value="Division 4.1"/>
      <xs:enumeration value="Division 4.2"/>
      <xs:enumeration value="Division 4.3"/>
      <xs:enumeration value="Division 5.1"/>
      <xs:enumeration value="Division 5.2"/>
      <xs:enumeration value="Division 6.1"/>
      <xs:enumeration value="Division 6.2"/>
      <xs:enumeration value="Class 1: Explosives"/>
      <xs:enumeration value="Class 1.1: Explosives"/>
      <xs:enumeration value="Class 1.2: Explosives"/>
      <xs:enumeration value="Class 1.3: Explosives"/>
      <xs:enumeration value="Class 1.4: Explosives"/>
      <xs:enumeration value="Class 1.5: Blasting Agents"/>
      <xs:enumeration value="Class 1.6: Explosives"/>
      <xs:enumeration value="Class 2.1: Flammable Gas"/>
      <xs:enumeration value="Class 2.2: Nonflammable Gas"/>
      <xs:enumeration value="Class 2.3: Poisonous Gas"/>
      <xs:enumeration value="Class 2.2: Oxygen (Alternative Placard)"/>
      <xs:enumeration value="Class 2.3: Inhalation Hazard (Alternative Placard)"/>
      <xs:enumeration value="Class 3: Flammable Liquids"/>
      <xs:enumeration value="Class 3: Combustible (Alternate Placard)"/>
      <xs:enumeration value="Class 3: Fuel Oil (Alternate Placard)"/>
      <xs:enumeration value="Class 3: Gasoline (Alternate Placard)"/>
      <xs:enumeration value="Class 4.1: Flammable Solids"/>
      <xs:enumeration value="Class 4.2: Spontaneously Combustible Solids"/>
      <xs:enumeration value="Class 4.3: Dangerous when Wet"/>
      <xs:enumeration value="Class 5.1: Oxidizing Agent"/>
      <xs:enumeration value="Class 5.2: Organic Peroxide Oxidizing Agent"/>
      <xs:enumeration value="Class 6.1: Poison"/>
      <xs:enumeration value="Class 6.2: Biohazard"/>
      <xs:enumeration value="Class 7: Radioactive"/>
      <xs:enumeration value="Class 8: Corrosive"/>
      <xs:enumeration value="Class 9: Miscellaneous"/>
    </xs:restriction>
  </xs:simpleType>
</xs:element>
```

Sources:
[1] https://en.wikipedia.org/wiki/GHS_hazard_pictograms
[2] https://en.wikipedia.org/wiki/Dangerous_goods#Classification_and_labeling_summary_tables

**Comment by macintoshpie** [comment]
@markborkum Thank you! 🎉

**Review by corymosiman12 — APPROVED** [review]
Looks good!

**Comment by corymosiman12** [comment]
@macintoshpie can you make sure we tracking this in L200 Audit schematron stuff and that once this merges we add necessary element checking if needed?

**Pull request closed without merging**